CDCgov/prime-simplereport
# Fixed 2 AccessDeniedException's

## Related Issue or Background Info

- https://app.zenhub.com/workspaces/simplereport-growth--engagement-605b43a6ceb92c000f86279a/issues/cdcgov/prime-simplereport/1365
- AccessDeniedException was raised when the same test is submitted twice (eg by two different users in short succession)
- AccessDeniedException was raised when entry only user tried to change patient's test result delivery preferences

## Changes Proposed

- Changed exception that's raised when a test result is submitted twice
- Changed permission restrictions on changing a patient's test result delivery preferences
- Added tests

## Additional Information

### UI
- [x] Any new or edited error messages have been approved by design (including @hmyers-cms or @katiealoisi to approve)

### Testing
- Verify that entry only users can change a patient's test result delivery preferences successfully
- Verify that loading the same test queue in two windows and trying to submit the same test in succession triggers an exception on the second submission that reads, "This test has already been submitted."

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.) (including @jeskew-gov to approve)
